compiler/rustc_data_structures/src/stable_hasher.rs

```diff
@@ -565,6 +565,12 @@ where
     }
 }
 
+impl_stable_traits_for_trivial_type!(::std::ffi::OsStr);
+impl_stable_traits_for_trivial_type!(::std::ffi::OsString);
+
+impl_stable_traits_for_trivial_type!(::std::ffi::CStr);
+impl_stable_traits_for_trivial_type!(::std::ffi::CString);
```

Comment:
Is it required to add the impls for `CStr(ing)` and `OsString` for `OsStr` to work?

Same question about the stuff in `compiler/rustc_middle/src/query/keys.rs`.

Reply:
Nope, I just thought I'd add them while I was at it. I can remove them if you think they're not desirable